### ISSUE TITLE
Replace hashtags with tags

### ISSUE THREAD

**Issue by hoodiek** [tootsuite/mastodon#2331]
hashtags as they are rarely used, and when they are used, they are clunky, and break the flow of a post as a thing to read, or just feel odd when in the body of it.

to alleviate this and improve on the feature, i suggest several changes

- [ ] making a separate section of the post for hashtags only, and an "add tag" button to let users choose from hashtags they have added before, and predefined ones such as #lewd or #art etc, as well as add new ones. the UI for this i can map out at a later time.
- [ ] making the ability to mute tags, presumably through the same feature as keyword muting, should it ever come into existence
- [ ] the ability to filter timelines (home, public and lists should they ever exist) to only show a tag, or hide tags from the timeline when viewing it normally (API side, as well as temporarily through UI in column settings)
- [ ] the ability to make images hidden by default / click to show if the post is tagged with tags you choose ('content-mute' option, somewhere)
- [ ] the ability to make your account default on all posts to have certain tags present on them
- [ ] make this extra section only be shown when you hover over the post or view it in full context in rightmost column
- [ ] hashtags added to the body of a post automatically added to section for tags only after the post
* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

**Comment by expenses** [comment]
I like this. the `CW` and `NSFW` buttons could be changed so that they're just shortcuts for adding those tags to a toot, which determines if its shown or not.

**Comment by msappir** [comment]
Another addition which could dovetail with this (and may have been mentioned in another issue as well) would be to have a setting for certain tags to show (unfold) CW'd text by default. This would reduce friction from using CWs for the sake of others on types of toots which one usually wants to read (e.g. I toot political stuff and want to read political stuff without having to click through every single time, if I could have all political toots shown by default I would have less reason not to CW my own political toots for the sake of others.

**Comment by Cassolotl** [comment]
> when they are used, they are clunky, and break the flow of a post as a thing to read, or just feel odd when in the body of it.

I like hashtags in sentences!

**Comment by hoodiek** [comment]
@Cassolotl there is nothing wrong with hashtags in sentences, i was staying that when you _can't_ fit them into the sentence, they are awkward and shoehorned onto either end, as such, for hashtags with no accurate place in the body text of the message, it makes sense to hide them in a seperate field that body-text-tags would also be duplicated in, to better make it legible and pretty, while retaining all current functionality

**Comment by Cassolotl** [comment]
Oh alright. :) Then I'm not opposed!

**Comment by 8zu** [comment]
This has the potential for even greater things. Some of additional idea

1. Let the client side automatically inject language hashtag to a post when it's published 
2. Have a way to link to a tag without making the post itself a hashtag (perhaps the syntax can be `#:hashtag` similar to the mediawiki syntax)
3. subscribe to a hashtag

**Comment by hoodiek** [comment]
all good ideas. there was already talk in the discord of subscribing to a hashtag, as that's essentially what GNUSocial "groups" do, allow you to follow a hashtag and put all posts to that hashtag into your timeline when posted, following the individual people or not

**Comment by raingloom** [comment]
@8zu that should be "hashtags" as I follow at least one person who posts in French and English at the same time. Don't assume a post is in just one language.

**Comment by sandhawke** [comment]
Also, opt-in hashtags.  You don't see the post at all unless you've chosen to see that hashtag.   Looking at someone's profile would show the opt-in hashtags they've activated.   I know this is a separate feature, but it seems to fit into this mix.   CW hashtags would then be semi-opt-in, I guess -- ask each time (until the user opts out (mute) or opts in (don't bother showing the warning).

**Comment by raingloom** [comment]
Tbh what's currently worst about hashtags is that you can't combine them in searches. Is there an open issue for that? If not I'll open one later.

**Comment by SoniEx2** [comment]
How does this handle accessibility? Currently you can just, as explained, shoehorn them at the end, without having to switch input fields.

**Comment by bunnybooboo** [comment]
@raingloom if you have time could throw that idea into a new issue? I like it and think it's deserving of it's own thread and implementation considerations.

**Comment by Openmedianetwork** [comment]
I like the ad-hock iline hashtags its the #openweb don't jail data into category if possible :)

**Comment by voronoipotato** [comment]
They can still be represented as hastags when federating, this way it doesn't break consistency with anyone we're federating with.

**Comment by SoniEx2** [comment]
This is an accessibility issue.

**Comment by voronoipotato** [comment]
honestly I'm not even sure how much value all this adds if any at all. Also SoniEx2 is right, it could introduce issues with accessibility and is additional work when devs already have 800 other issues right now.

**Comment by ThisIsMissEm** [comment]
I feel like the title of this issue is misleading, as the OP is not suggesting removing hashtags, but rather, allowing hashtags to be added and shown separately from a Toot.

I think we could implement this purely in client-side, especially if we exposed entities information via the API for web (or we could just regex it / replicate the text parser that's used on the backend), then we could find the hashtags that are grouped at the end of a toot's body and collapse it.

**Comment by trwnh** [comment]
From my understanding, the intent of this issue is to make tags more like Tumblr and less like Twitter, right? If so, I feel it does somewhat complicate the usage of tags (which can be inserted anywhere, as opposed to simply appended at the bottom). However, suggesting previously-used tags is not a bad idea.

With that said, I'd like to raise a small tangent to discuss the more taxonomical use of this proposed tagging system. Semantically, treating tags as metadata and not simply hyperlinks is more of a way of organizing a post, rather than as a way to surface other posts (although either system can be used for both). If we are trying to simply categorize our posts without necessarily surfacing them, how would we do that? If we are trying to surface our posts without necessarily categorizing them, how would we do that?
